### DISCUSSION TITLE
Playlist queue

### DISCUSSION THREAD

**Discussion by Snooze1000** [punxaphil/custom-sonos-card#109] [Q&A] [answered]
First, thanks for this great development which I used to create a kids specific dashboard to control a sonos speaker in the child's room.

I just wanted to ask if there is also a possibility to integrate the sonos queue/playlist?

**Comment by punxaphil** [comment]
Thanks for the kind words. Unfortunately I haven’t found any support for that in the Sonos integration, which is a must before I can show anything in the card 😞

**Comment by psjnr84** [comment]
Hi Johan - I second what Snooze said, great piece of custom work on this. Yes, the standard Sonos Integration doesn't seem to support that, but I've created something in Node Red that allows for that using some sonos events and customplus integrations to do that very thing. I then use iframes to display a webpart to show the list / queue for a given player and that allows the user to see what's currently playing and then select the item to playback on the relevant speaker. Do you think this type of integration could be integrated into your custom card and ultimately provide everyone with the best experience?

**Reply by punxaphil** [reply]
Honestly, I have no idea. Sounds advanced. Feel free to play around with it though!

**Comment by punxaphil** [comment, marked as answer]
@Snooze1000 @psjnr84 
There is now queue/playlist added in the beta: https://github.com/punxaphil/custom-sonos-card/issues/355#issuecomment-2440158692

**Reply by psjnr84** [reply]
Fantastic guys. This was the final part of the jigsaw for your card and can confirm the queue works really well. Thanks for implementing this with the latest release and allowing me to fully transition to the card without needing other items / workarounds I’ve created to do this outside of your card.

**Reply by punxaphil** [reply]
Great you like it! ❤️ 


---
PunxaPhil
[![image](https://github.com/user-attachments/assets/cb2e240e-f9ca-4900-a425-b8089216c1ee)](https://patreon.com/PunxaPhil) [![image](https://github.com/johanfrick/custom-sonos-card/assets/26582898/9bebc1f6-6bda-4cbd-b596-50d3d2f712de)](https://www.buymeacoffee.com/punxaphil)

**Reply by psjnr84** [reply]
Another question. Under the browse sources, and Spotify option, do you know if there could ever be a way to allow you to search for a track or album from inside your card?

Lastly, do you know if you can customise the sources in the list? It would be great to only list certain pertinent sources shown instead of showing all media types.

**Reply by punxaphil** [reply]
You mean "browse media"? That is implemented by the HA team. You need to ask them to add support for sorting and searching.
